sympy/functions/special/bessel.py

```diff
@@ -216,11 +216,46 @@ def _eval_rewrite_as_bessely(self, nu, z, **kwargs):
     def _eval_rewrite_as_jn(self, nu, z, **kwargs):
         return sqrt(2*z/pi)*jn(nu - S.Half, self.argument)
 
+    def _eval_as_leading_term(self, x, cdir=0):
+        nu, z = self.args
+        arg = z.as_leading_term(x)
+        if x in arg.free_symbols:
+            return z**nu
+        else:
+            return self.func(*self.args)
+
     def _eval_is_extended_real(self):
         nu, z = self.args
         if nu.is_integer and z.is_extended_real:
             return True
 
+    def _eval_nseries(self, x, n, logx, cdir=0):
+        from sympy.series.order import Order
+        nu, z = self.args
+
+        # In case of powers less than 1, number of terms need to be computed
+        # separately to avoid repeated callings of _eval_nseries with wrong n
+        try:
+            _, exp = z.leadterm(x)
+        except (ValueError, NotImplementedError):
+            return self
+
+        if exp.is_positive:
+            newn = ceiling(n/exp)
+            o = Order(x**n, x)
+            r = (z/2)._eval_nseries(x, n, logx, cdir).removeO()
+            t = ((r**2).expand() + o).removeO()
+
+            term = 1/gamma(nu + 1)
```

Comment:
Could this be `term = r**nu/gamma(nu + 1)`?

Reply:
Oh, that is a nice change, more terms would be truncated this way. Will make the change.

I am concerned about the case where n is given a small value, say 1, in which case r would be 0, and if `nu` is a negative non-integer, `r**nu` evaluates to `zoo`. In the case of `r = 0`, should I add a condition that would simply return `o`? (`o = O(x**n, x`)
Or would any other value be more appropriate to return?

Reply:
>  that would simply return o

That looks good to me.

Reply:
Also consider this: https://github.com/sympy/sympy/pull/21049#issuecomment-797616965